src/test/java/net/openhft/chronicle/bytes/MappedBytesEdgeTest.java

```diff
@@ -45,80 +48,93 @@ public MappedBytesEdgeTest(int size, ReadWrite rw, Consumer<Bytes<?>> doit) {
+                {6, ReadWrite.WRITE, (Consumer<Bytes<?>>) (b -> b.write8bit("hello"))},
+                {7, ReadWrite.WRITE, (Consumer<Bytes<?>>) (b -> b.appendUtf8("doggie"))},
+                {10, ReadWrite.WRITE, (Consumer<Bytes<?>>) (b -> b.write(Bytes.from("armadillo")))},
 
         });
     }
 
     @Test
     public void testCorrectChunkResolved() throws IOException {
         final File tempMBFile = Files.createTempFile("mapped", "bytes").toFile();
-        int chunk = 262144;
-        int overlap = 65536;
-        try (final MappedBytes bytes = MappedBytes.mappedBytes(tempMBFile, chunk, overlap)) {
+        final long overlap = OS.mapAlign(overlap(CHUNK_SIZE));
+        if (overlap != overlap(CHUNK_SIZE)) {
+            Jvm.warn().on(MappedBytesEdgeTest.class, "You configured an invalid overlap " + overlap(CHUNK_SIZE) + ", actual one used will be " + overlap);
+        }
+        try (final MappedBytes bytes = MappedBytes.mappedBytes(tempMBFile, CHUNK_SIZE, overlap)) {
             // map in the real file
             bytes.writePosition(0).writeByte((byte) 0);
             assertEquals(0, bytes.bytesStore().start());
 
             if (rw == ReadWrite.WRITE) {
-                checkWritePosition(bytes, chunk + overlap - size, 0);
-                checkWritePosition(bytes, chunk + overlap, chunk);
-                checkWritePosition(bytes, chunk + overlap + size, chunk);
+                checkWritePosition(bytes, CHUNK_SIZE + overlap - size, 0);
+                checkWritePosition(bytes, CHUNK_SIZE + overlap, CHUNK_SIZE);
+                checkWritePosition(bytes, CHUNK_SIZE + overlap + size, CHUNK_SIZE);
                 // go back to just before the overlap - will still be in second chunk
-                checkWritePosition(bytes, chunk + overlap - size, chunk);
-                checkWritePosition(bytes, chunk - size, 0);
+                checkWritePosition(bytes, CHUNK_SIZE + overlap - size, CHUNK_SIZE);
+                checkWritePosition(bytes, CHUNK_SIZE - size, 0);
+                // now try and write over the end of the chunk
+                checkWritePosition(bytes, CHUNK_SIZE - 1, 0);
+                // and end of chunk plus offset
+                checkWritePosition(bytes, CHUNK_SIZE + overlap - 1, size > 1 ? CHUNK_SIZE : 0);
+
                 if (size > 1) {
-                    // now try and write over the end of the chunk
-                    bytes.writePosition(chunk - 1);
-                    doit.accept(bytes);
-                    // and end of chunk plus offset
-                    bytes.writePosition(chunk + overlap - 1);
-                    doit.accept(bytes);
+                    // go back to the first store
+                    checkWritePosition(bytes, CHUNK_SIZE - 1, 0);
+                    // write to just before the start of the third bytes store (i.e. not in the 1st plus overlap or 2nd plus overlap)
+                    checkWritePosition(bytes, 3 * CHUNK_SIZE - size + 1, 2 * CHUNK_SIZE);
```

Comment:
This reproduces the bug. If you pop the last commit (with the fix in it) off, you will see it blow.